### ISSUE TITLE
require comment explaining `type:ignore` comments

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#571]
incorrect:
```py
def foo():
    a: str = 1  # type:ignore[assignment] # error: add a comment on the line above explaining why the error is being ignored
```

correct:
```py
def foo():
    # false positive, see https://github.com/KotlinIsland/basedmypy/issues/252
    a: str = 1  # type:ignore[assignment]
```

eslint has a rule for this: https://typescript-eslint.io/rules/ban-ts-comment/

**Comment by Zeckie** [comment]
Does it need to be the line immediately above the ignored error, or can it be the line above the statement?

And will another `# type: ignore` comment (or something similar, like a `# noqa` comment) count?

**Comment by KotlinIsland** [comment]
I can see many options here:
```py
1 + ""  # type: ignore[operator]
```
Which would report
```
test.py:1 Type ignore without reason  [ignore-reason]
```

then you could just suppress that:
```
a + ""  # type: ignore[operator, ignore-reason]
```

but to properly handle it maybe:
```
# blah blah
1 + ""   # type: ignore[operator]
OR
1 + ""  # type: ignore[]operator] blah blah
```

if wewanted to get very specific, it could have a separate pragma, to avoid accidentaly appeasing the rule
```
# mypy-comment: blah blah
1 + ""  # type: ignore[operator]
```
that seems really wordy though.

I feel that the comment on the line above is nicer, but prone to accident(n`noqa` etc). after the ignore feels more sound, but more clunky, so i'm not sure
